app/views/layouts/application.html.erb

```diff
@@ -9,19 +9,35 @@
     <%= stylesheet_link_tag "application", "data-turbo-track": "reload" %>
     <%= javascript_importmap_tags %>
   </head>
+  <nav>
+    <ul>
+      <li>
+        <%= link_to "Pets", "/pets" %>
+      </li>
+      <li>
+        <%= link_to "Shelters", "/shelters" %>
+      </li>
+      <li>
+        <%= link_to "Veterinarians", "/veterinarians" %>
+      </li>
+      <li>
+        <%= link_to "Veterinary Offices", "/veterinary_offices" %>
+      </li>
+      <li>
+        <%= link_to 'Admin Shelters', admin_shelters_path, data: { turbo_method: :get } %>
+      </li>
+      <li>
+        <%= link_to 'Admin Applications', admin_applications_path, data: { turbo_method: :get } %>
+      </li>
+    </ul>
 
+  </nav>
   <body>
-  <% flash.each do |type, msg| %>
-    <div>
-      <%= msg %>
-    </div>
-  <% end %>
-    <section>
-      <p><%= link_to "Pets", "/pets" %></p>
-      <p><%= link_to "Shelters", "/shelters" %></p>
-      <p><%= link_to "Veterinarians", "/veterinarians" %></p>
-      <p><%= link_to "Veterinary Offices", "/veterinary_offices" %></p>
-    </section>
-    <%= yield %>
+    <% if !params[:pet_name] %>
```

Comment:
I may need to play around with your code more, and you know it closer than I do, but I'm tempted to think that you don't need this conditional to only render the flahs if there's no pet_name. You should only be adding a flash message when there is an error case (i.e. when the pet name isn't found), so I believe you should be able to render this flash layout no matter what, and the message will just show up on the page if there is a message available. 

You would have to take it out of your application show view though, as you'll only want to render it once on the page

Reply:
Ahhh I see! For the different placement, this makes sense! There are definitely many different ways to do this with various FE tools, but now that I understand that the 2 flash messages are distinct, I understand the need to have the if statement here!